### PR TITLE
Remove auto increment from contribution_queue

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/8804

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

- Create new profile
- Enable Rewards
- Ensure that `contribution_queue` table id column is of type `TEXT` in database
- Ensure that `contribution_queue_publishers` table id column is of type `TEXT` in database
- Tip a publisher and ensure that it is processed and summary looks as expected

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc [hunk 1]
```diff
@@ -164,7 +215,7 @@ void DatabaseContributionQueue::InsertOrUpdate(
 void DatabaseContributionQueue::OnInsertOrUpdate(
     ledger::DBCommandResponsePtr response,
     const std::string& queue_string,
-    const uint64_t id,
+    const std::string& id,
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc:218]: I think that id is not needed anymore, as you assign `info->id` all the time now in Insert function, so you should have it in `queue_string`

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc [hunk 2]
```diff
@@ -99,6 +125,51 @@ bool DatabaseContributionQueue::MigrateToV15(
...
+    return false;
+  }
+
+  if (!CreateTableV23(transaction)) {
+    BLOG(0, "Table couldn't be created");
+    return false;
+  }
+
+  // Migrate the contribution_queue table
+  std::string query = base::StringPrintf(
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc:147]: nit: const

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue_publishers.cc [hunk 3]
```diff
@@ -170,11 +207,66 @@ bool DatabaseContributionQueuePublishers::MigrateToV15(
...
+  const std::string temp_table_name = base::StringPrintf(
+      "%s_temp",
+      kTableName);
+
+  if (!RenameDBTable(transaction, kTableName, temp_table_name)) {
+    BLOG(0, "Table couldn't be renamed");
+    return false;
+  }
+
+  std::string query =
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue_publishers.cc:223]: nit: const

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc (hunk 1 above)
Comment (emerick) [reply vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc]: Good catch, fixed.

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc (hunk 2 above)
Comment (emerick) [reply vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc]: Fixed.

**Pull request merged**